docs/source/daml/intro/5_Restrictions.rst

```diff
@@ -0,0 +1,103 @@
+
+On a distributed DAML ledger, there are no guarantees that ledger effective time or relative time are strictly increasing. The only guarantee is that ledger effective time is increasing with causality. If a transaction ``TX2`` depends on a transaction ``TX1``, then the ledger enforces that the LET of ``TX2`` is greater than or equal to that of ``TX1``:
+
+.. literalinclude:: daml/Intro_5_Restrictions.daml
+  :language: daml
+  :start-after: -- CAUSALITY_TEST_BEGIN
+  :end-before: -- CAUSALITY_TEST_END
+
+Actions and ``do`` blocks
+-------------------------
+
+You have come across ``do`` blocks and ``<-`` notations in two contexts by now: ``Scenario`` and ``Update``. Both of these are examples of an ``Action``, also called a *Monad* in functional programming.
+
+Expressions in DAML are pure in the sense that they have no side-effects. They neither read nor modify any external state. If you know the value of all variables in scope and write an expression, you can work out the value of that expression on pen and paper. All the expressions you've seen that used the ``<-`` notation were not like that.
+
+``getTime`` is a good example. There is nothing in scope that could inform the value of ``now`` so there is no expression ``expr`` that you could put on the right hand side of ``now = expr``. To get the ledger effective time, you need to be in the context of a submitted transaction and then look at that context.
+
+Similarly, you've come across ``fetch``. If you have ``cid : ContractId Account`` in scope and you come across the expression ``fetch cid``, you can't evaluate that to an ``Account`` so you can't write ``account = fetch cid``. To do so, you'd have to have a ledger on scope on which to look up that contract id.
+
+Actions are a way to handle such "impure" expressions. ``Action a`` is a type class with a single parameter ``a`` and ``Update`` and ``Scenario`` are instances of ``Action``. A value of such a type ``m a`` where ``m`` is an instance of ``Action`` can be interpreted as "a recipe for an action of type ``m``, which, when executed, returns a value ``a``".
+
+- An ``Update a`` is "a recipe to update a DAML ledger, which, when committed, returns a value of type ``a``". An update to a DAML ledger is a transaction so equivalently, an ``Update a`` is "a recipe to construct a transaction, which, when executed in the context of a ledger, returns a value of type ``a``".
+- A ``Scenario a`` is "a recipe to submit transactions to a test ledger, which, when performed against a ledger, returns a value of type ``a``".
+
+Expressions like ``getTime``, ``getParty``, ``pass``, ``submit``, ``create`` and ``exercise`` should make more sense in that light. For example:
```

Comment:
Thought it might be a bit confusing to say expressions, as above you've contrasted Actions with expressions.
```suggestion
Functions like ``getTime``, ``getParty``, ``pass``, ``submit``, ``create`` and ``exercise`` should make more sense in that light. For example:
```